js/quote-generator.js

```diff
@@ -0,0 +1,112 @@
+const randomBtn = document.querySelector("#random-btn");
+const allAuthorsBtn = document.querySelector("#all-authors-btn");
+const backToQuotesBtn = document.querySelector("#back-to-quotes-btn");
+const quote = document.querySelector(".quote");
+const quoteAuthor = document.querySelector(".quote-author");
+const quoteGenre = document.querySelector(".quote-genre");
+const allAuthorsContainer = document.querySelector(".all-authors-container");
+const gridContainer = document.querySelector(".grid-container");
+const topButtons = document.querySelector(".top-buttons");
+const footer = document.querySelector("footer");
+const quoteLoader = document.querySelector("#quote-loader");
+const allAuthorsLoader = document.querySelector("#all-authors-loader");
+const topTenCaption = document.querySelector(".top-10-caption");
+const baseUrl = "https://api.quotable.io";
+const randomQuoteUrl = new URL("/random", baseUrl);
+const allAuthorsUrl = new URL(
+  "/authors?sortBy=quoteCount&order=desc&limit=10",
+  baseUrl
+);
+
+let allAuthorsFetchCalled = false;
+
+async function getRandomQuote() {
+  try {
+    quoteLoader.style.display = "flex";
+    quote.style.display = "none";
+    quoteAuthor.style.display = "none";
+    quoteGenre.style.display = "none";
+    const promise = await fetch(randomQuoteUrl);
+    if (!promise.ok) {
+      throw new Error(`HTTP error! Status: ${promise.status}`);
+    }
+    quoteLoader.style.display = "none";
+    quote.style.display = "block";
+    quoteAuthor.style.display = "block";
+    quoteGenre.style.display = "block";
+    return (result = await promise.json());
+  } catch (error) {
+    console.error("Error:", error);
+    quoteLoader.style.display = "none";
+    quote.style.display = "block";
+    quote.style.color = "red";
+    quote.textContent =
+      "An error occurred while generating your quote. Please try again later.";
+    throw error;
+  }
+}
+
+async function setRandomQuote() {
+  const { content, author, tags } = await getRandomQuote();
+  quote.textContent = content;
+  quoteAuthor.textContent = author;
+  quoteGenre.textContent = tags;
+}
+
+randomBtn.addEventListener("click", setRandomQuote);
+setRandomQuote();
+
+async function getAllAuthors() {
+  try {
+    allAuthorsLoader.style.display = "flex";
+    const promise = await fetch(allAuthorsUrl);
+    if (!promise.ok) {
+      throw new Error(`HTTP error! Status: ${promise.status}`);
+    }
+    allAuthorsLoader.style.display = "none";
+    return (result = await promise.json());
```

Comment:
there is no need to create variable `result` 

you can do just `return promise.json()`

or `return await promise.json()`

By the way, do you know the difference between:

`return await promise.json()` and `return promise.json()`

both variants will work, but here is a little difference, try to google